app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -77,14 +77,36 @@ class SectionLink {
     return this.name === undefined || this.validName ? "" : "warning";
   }
 
+  get internal() {
+    return (
+      this.value.startsWith(this.protocolAndHost) || this.value.startsWith("/")
```

Comment:
What if someone used `http` instead of `https` for a site's URL? Would we end up considering that external?

Reply:
good catch, I will fix it to check http|https + window.location.host